src/OnboardingSPA/components/NewfoldInterfaceSkeleton/index.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+/**
+ * WordPress dependencies
+ */
+import { forwardRef, useEffect } from '@wordpress/element';
+// eslint-disable-next-line  @wordpress/no-unsafe-wp-apis
+import { __unstableUseNavigateRegions as useNavigateRegions } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useMergeRefs } from '@wordpress/compose';
+
+function useHTMLClass( className ) {
+	useEffect( () => {
+		const element =
+			document && document.querySelector( `html:not(.${ className })` );
+		if ( ! element ) {
+			return;
+		}
+		element.classList.toggle( className );
+		return () => {
+			element.classList.toggle( className );
+		};
+	}, [ className ] );
+}
+
+function NewfoldInterfaceSkeleton(
+	{
+		footer,
+		header,
+		sidebar,
+		secondarySidebar,
+		notices,
+		content,
+		drawer,
+		actions,
+		labels,
+		className,
+		shortcuts,
+	},
+	ref
+) {
+	const navigateRegionsProps = useNavigateRegions( shortcuts );
+
+	useHTMLClass( 'nfd-interface-interface-skeleton__html-container' );
+
+	const defaultLabels = {
+		/* translators: accessibility text for the nav bar landmark region. */
+		drawer: __( 'Drawer' ),
+		/* translators: accessibility text for the top bar landmark region. */
+		header: __( 'Header' ),
+		/* translators: accessibility text for the content landmark region. */
+		body: __( 'Content' ),
+		/* translators: accessibility text for the secondary sidebar landmark region. */
+		secondarySidebar: __( 'Block Library' ),
+		/* translators: accessibility text for the settings landmark region. */
+		sidebar: __( 'Settings' ),
+		/* translators: accessibility text for the publish landmark region. */
+		actions: __( 'Publish' ),
+		/* translators: accessibility text for the footer landmark region. */
+		footer: __( 'Footer' ),
+	};
+
+	const mergedLabels = { ...defaultLabels, ...labels };
+
+	return (
+		<div
+			{ ...navigateRegionsProps }
+			ref={ useMergeRefs( [ ref, navigateRegionsProps.ref ] ) }
+			className={ classnames(
+				className,
+				'nfd-interface-interface-skeleton',
+				navigateRegionsProps.className,
+				!! footer && 'has-footer'
+			) }
+		>
+			{ !! drawer && (
```

Comment:
Can we just check why `!!` was used?

Reply:
The double exclamation point, or double bang, converts a truthy or falsy value to “true” or “false”. In other words, it operates exactly like Boolean(value) would. [Ref](https://medium.com/@chirag.viradiya_30404/whats-the-double-exclamation-sign-for-in-javascript-d93ed5ad8491#:~:text=The%20double%20exclamation%20point%2C%20or,like%20Boolean(value)%20would.)

Reply:
Thanks